package-system/AWSGameLiftServerSDK/build_package_image.py

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import json
+import os
+import pathlib
+import subprocess
+import shutil
+import urllib.request
+import zipfile
+from typing import List
+
+PACKAGE_NAME: str = "AWSGameLiftServerSDK"
+PACKAGE_VERSION: str = "3.4.2-rev1"
+PACKAGE_PLATFORM: str = ""
+PACKAGE_URL: str = "https://aws.amazon.com/documentation/gamelift/"
+PACKAGE_LICENSE: str = "Apache-2.0"
+PACKAGE_LICENSE_FILE: str = "LICENSE_AMAZON_GAMELIFT_SDK.TXT"
+
+GAMELIFT_SERVER_SDK_RELEASE_VERSION: str = "4.0.2"
+GAMELIFT_SERVER_SDK_DOWNLOAD_URL: str = "https://gamelift-release.s3-us-west-2.amazonaws.com/GameLift_06_03_2021.zip"
+
+PACKAGE_BASE_PATH: pathlib.Path = pathlib.Path(os.path.dirname(__file__))
+PACKAGE_ROOT_PATH: pathlib.Path = PACKAGE_BASE_PATH.parent
+PACKAGE_BUILD_TYPES: List[str] = ["Debug", "Release"]
+PACKAGE_LIB_TYPES: List[str] = ["Shared", "Static"]
+PACKAGE_PLATFORM_OPTIONS: List[str] = ["windows", "linux"]
+
+# utils
+class WorkingDirectoryInfo(object):
+    def __init__(self, root_path: pathlib.Path,
+                 source_path: pathlib.Path,
+                 build_path: pathlib.Path,
+                 output_path: pathlib.Path,
+                 libs_output_path: pathlib.Path) -> None:
+        self.root_path = root_path
+        self.source_path = source_path
+        self.build_path = build_path
+        self.output_path = output_path
+        self.libs_output_path = libs_output_path
+
+def subp_args(args) -> str:
+    arg_string = " ".join([arg for arg in args])
+    print(f"Command: {arg_string}")
+    return arg_string
+
+def delete_folder(folder: pathlib.Path) -> None:
+    shutil.rmtree(folder.resolve())
+
+# steps
+def collect_package_info() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--platform-name',
+        required=True,
+        choices=PACKAGE_PLATFORM_OPTIONS
+    )
+    
+    args = parser.parse_args()
+    global PACKAGE_PLATFORM
+    PACKAGE_PLATFORM = args.platform_name
+
+def prepare_working_directory() -> WorkingDirectoryInfo:
+    # root directory
+    root_directory: pathlib.Path = PACKAGE_BASE_PATH.joinpath("temp")
```

Comment:
Nit: May worth making the temp path a CLI argument people can specify in the configuration file instead of hardcoding it here.

Reply:
this is the default behavior of build from git script